### PR TITLE
aerosol init add docs and tests

### LINKED ISSUE #823: improve docstrings for DryAerosolMixture

### LINKED ISSUE #826: add unit tests for aerosol_composition

### PR DESCRIPTION
**Pull request by claresinger**
closes #823 and closes #826 

more ideas for tests to add always appreciated!

### CODE REVIEW

File: tests/unit_tests/initialisation/test_aerosol_init.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+@pytest.mark.parametrize(
+    "mass_fractions",
+    [
+        pytest.param({"(NH4)2SO4": 1.0, "insoluble": 0.0}),
+        pytest.param({"(NH4)2SO4": 0.5, "insoluble": 0.5}),
+        pytest.param({"(NH4)2SO4": 0.0, "insoluble": 1.0}),
+    ],
+)
+def test_aerosol_init(mass_fractions):
```
Comment (slayoo) [inline tests/unit_tests/initialisation/test_aerosol_init.py:21]: perhaps let's rename the test to something more descriptive like `test_volume_weighted_kappa_with_insoluble_compound()`?

File: tests/unit_tests/initialisation/test_aerosol_init.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        is_soluble=is_soluble,
+        ionic_dissociation_phi=ionic_dissociation_phi,
+    )
+
+    # Act
+    kappa = aer.kappa(mass_fractions)
+    volume_fractions = aer.volume_fractions(mass_fractions)
+
+    # Assert
+    kappas = {"(NH4)2SO4": 0.72, "insoluble": 0.0}
```
Comment (slayoo) [inline tests/unit_tests/initialisation/test_aerosol_init.py:48]: having both `kappa` and `kappas` within one block, and the `kappa` not being one of `kappas` seems misleading, perhaps let's do `kappa_expected = aer.kappa(mass_fractions)['Constant']` and `kappa_actual=...`

**Comment by claresinger** [comment]
@slayoo do you know why some of the (not new) tests are failing? specifically it seems to be this one:
```
tests/smoke_tests/arabas_and_shima_2017/test_vs_bdf.py EEEEEEEEEEEEEEEEE [ 50%]
```

**Comment by slayoo** [comment]
> @slayoo do you know why some of the (not new) tests are failing? specifically it seems to be this one:
> 
> ```
> tests/smoke_tests/arabas_and_shima_2017/test_vs_bdf.py EEEEEEEEEEEEEEEEE [ 50%]
> ```
> 
> update: this tests seems to pass locally, so now I'm even more confused. could it be a test-time-req issue? I thought so but it seems like it's already pointing to a new enough version of PySDM-examples.
 
The BDF tests have a bad habit o failing with internal SciPy errors only on CI:
- https://github.com/atmos-cloud-sim-uj/PySDM/issues/577
- https://github.com/atmos-cloud-sim-uj/PySDM/issues/293
I have never been able to reproduce these errors locally!

**Pull request merged**